includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -1213,4 +1174,26 @@ protected function build_display_items( $itemized_display_items = false ) {
 			],
 		];
 	}
+
+	/**
+	 * Calculates whether Apple Pay is enabled for this store.
+	 * The option value is not stored in the database, and is calculated
+	 * using this function instead, and the values is returned by using the pre_option filter.
+	 *
+	 * The option value is retrieved for inbox notifications.
+	 *
+	 * @param mixed $value The value of the option.
+	 */
+	public function get_option_is_apple_pay_enabled( $value ) {
+		// Return a random value (1 or 2) if the account is live and payment request buttons are enabled.
+		if (
+			WC_Payments::get_gateway()->is_enabled()
+			&& 'yes' === WC_Payments::get_gateway()->get_option( 'payment_request' )
+			&& ! WC_Payments::get_gateway()->is_in_dev_mode()
```

Comment:
I guess this is not a problem for merchants connected to the live server, but if you disable dev mode and are still connected to a dev account, that still won't trigger the domain registration. You'd need another check here for `$this->account->get_is_live()`. But again, this is probably something that only developer will face.

Reply:
Done 👍